File: backend/src/main/java/com/englishlearning/backend/service/MembershipPackageService.java
package com.englishlearning.backend.service;


import com.englishlearning.backend.dto.request.MembershipPackageCreateRequest;
import com.englishlearning.backend.dto.request.MembershipPackageUpdateRequest;
import com.englishlearning.backend.dto.response.MembershipPackageResponse;

import java.util.List;

public interface MembershipPackageService {

    List<MembershipPackageResponse> getAll();

    List<MembershipPackageResponse> getActivePackages();

    MembershipPackageResponse getById(Long id);

    MembershipPackageResponse create(
            MembershipPackageCreateRequest request
    );

    MembershipPackageResponse update(
            Long id,
            MembershipPackageUpdateRequest request
    );

    void delete(Long id);

    MembershipPackageResponse deactivate(Long id);

    MembershipPackageResponse activate(Long id);
}

File: frontend/src/services/membershipPackageService.js
import axiosClient from "../api/axiosClient";

const membershipPackageService = {
  getActivePackages() {
    return axiosClient.get("/v1/membership-packages/active");
  },
  getAll() {
    return axiosClient.get("/v1/membership-packages");
  },

  getById(id) {
    return axiosClient.get(`/v1/membership-packages/${id}`);
  },

  create(data) {
    return axiosClient.post("/v1/membership-packages", data);
  },

  update(id, data) {
    return axiosClient.put(`/v1/membership-packages/${id}`, data);
  },

  delete(id) {
    return axiosClient.delete(`/v1/membership-packages/${id}`);
  },

  deactivate(id) {
    return axiosClient.put(`/v1/membership-packages/${id}/deactivate`);
  },

  activate(id) {
    return axiosClient.put(`/v1/membership-packages/${id}/activate`);
  },
};

export default membershipPackageService;
